src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/NuPickersValueConnector.cs

```diff
@@ -0,0 +1,198 @@
+            JSON,
+            XML,
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuPickersValueConnector"/> class.
+        /// </summary>
+        /// <param name="entityService">An <see cref="IEntityService"/> implementation.</param>
+        public NuPickersValueConnector(IEntityService entityService)
+        {
+            _entityService = entityService ?? throw new ArgumentNullException(nameof(entityService));
+        }
+
+        public IEnumerable<string> PropertyEditorAliases => Enumerable.Empty<string>();
+
+        /// <summary>
+        /// Gets the deploy property corresponding to a content property.
+        /// </summary>
+        /// <param name="property">The content property.</param>
+        /// <param name="dependencies">The content dependencies.</param>
+        /// <returns>The deploy property value.</returns>
+        public string GetValue(Property property, ICollection<ArtifactDependency> dependencies)
+        {
+            // get the property value
+            var value = property.Value as string;
+
+            // parse the value - checking the format - CSV, XML or JSON
+            var items = ParseValue(value, out SaveFormat format);
+
+            var result = new List<KeyValuePair<string, string>>();
+
+            // loop over the values
+            foreach (var item in items)
+            {
+                if (int.TryParse(item.Key, out int nodeId))
+                {
+                    // if an INT, attempt to get the UDI
+                    var guidUdi = GetGuidUdi(nodeId);
+                    if (guidUdi != null)
+                    {
+                        dependencies.Add(new ArtifactDependency(guidUdi, false, ArtifactDependencyMode.Exist));
+                        result.Add(new KeyValuePair<string, string>(guidUdi.ToString(), item.Value));
+                    }
+                    else
+                    {
+                        // the value isn't a UDI, assume it's "just a string"
+                        result.Add(item);
+                    }
+                }
+                else
+                {
+                    // the value isn't a node ID, assume it's "just a string"
+                    result.Add(item);
+                }
+            }
+
+            // return the value in the correct format - CSV, XML, JSON, (whatevs)
+            return SerializeValue(result, format);
+        }
+
+        /// <summary>
+        /// Sets a content property value using a deploy property.
+        /// </summary>
+        /// <param name="content">The content item.</param>
+        /// <param name="alias">The property alias.</param>
+        /// <param name="value">The deploy property value.</param>
+        public void SetValue(IContentBase content, string alias, string value)
+        {
+            // parse the value - checking the format - CSV, XML or JSON
+            var items = ParseValue(value, out SaveFormat format);
```

Comment:
Small detail - if you're amending the PR anyway - could you do the old (non C#6) syntax for the `out`'s (there's a few more). I'm not sure if all our build stuff is completely happy about this yet .. so I usually just try to avoid it for a little bit longer until we're completely sure it works 100% .. I know, I know.. minor detail .. but yeah.. funky stuff has been happening as we switched from AppVeyor to VSTS for our builds 😄

Reply:
No worries, I'll fix it up. I hit the same problem on Ditto recently, turned out to be we were using an older MSBuild version.

Reply:
Yeah I think it was the same issue here - we might have it fixed.. I'm not totally sure though so just to be safe for a while